src/video-player.js

```diff
@@ -336,6 +362,10 @@ class VideoPlayer extends Utils.mixin(Eventable) {
     if (VideoPlayer.allowUsageReport()) {
       options.usageReport = true;
     }
+    this.nbCalls = 0;
```

Comment:
You should also use `clearTimeout` on the `retryVideo` timeout before reseting nbCalls. Otherwise, it may still affect `nbCalls` count for the next source you're setting for the video.